routing/pathfind.go

```diff
@@ -494,36 +486,60 @@ func findPath(tx *bolt.Tx, graph *channeldb.ChannelGraph,
 		return nil, err
 	}
 
-	// We'll also include all the nodes found within the additional edges
-	// that are not known to us yet in the distance map.
-	for vertex := range additionalEdges {
+	additionalEdgesWithSrc := make(map[Vertex][]*edgePolicyWithSource)
+	for vertex, outgoingEdgePolicies := range additionalEdges {
```

Comment:
I can make it indexed by target node, but then I completely loose the source of the channel edge. Then the source node is not in ChannelEdgePolicy and neither in the additionalEdges map.

Reply:
you can just make `additionalEdges` contain `edgePolicyWithSource` elements :)

Reply:
In that case, I'd not pass it in as a map anymore. It would contain redundant info (edgePolicyWithSource.source == map keys). I could do it as a list, but I'd still need to create a lookup table out of it. And also, it feels like an implementation detail of this specific algorithm to me. In another algorithm we might need it in a different shape again. 

For me, map[Vertex][]*channeldb.ChannelEdgePolicy looks like the most natural way to provide the info, from the caller point of view.

If we don't want to do the reverse lookup table, I would prefer to add the source node to ChannelEdgePolicy. But this is probably a wide spread change. 

In short, I think it is acceptable to keep it like it is and accept that we now have a edgePolicyWithSource as a routing algo internal data type.

Reply:
Indeed, if we would add the source node as a member to `ChannelEdgePolicy`, the target node (`Node`) could be used as map key. But then still the question remains whether this isn't a implementation detail that the caller shouldn't be concerned with when building the `additionalEdges` map?

Reply:
When `ChannelEdgePolicy` contains both source and target nodes, it would be better (no redundant info) to pass additionalEdges as a list instead of a map. But then a conversion to map inside findPath would still be needed.

Reply:
Yeah, we probably shouldn't change `ChannelEdgePolicy`, as it directly maps to the wire messages being sent on the network. I'm fine with keeping this as is now, as I agree that this could be considered an implementation detail the caller shouldn't need to worry about.